### PR TITLE
feat: ignore IMMUTABLE fields in update tests

### PR DESCRIPTION
**Pull request by radhus**

### CODE REVIEW

**Comment by radhus** [comment]
> should they be ignored, or should that be check that they are immutable? i.e try to update them

It is a good idea, but I think that is out of scope for this PR as it would require some bigger changes.

The only input the test suite has is the result of the `Update()` function, which returns a valid updated object (i.e. no immutables should be modified). The test you propose would need to set the immutable field to a reasonable value (which can be of any type). I don't think there is an easy way to accomplish this, what do you think? If not, I think it is a nice idea to pursue further on!

File: internal/aiptest/update/required_fields.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ var requiredFields = suite.Test{
 	),
 	Generate: func(f *protogen.GeneratedFile, scope suite.Scope) error {
 		updateMethod, _ := util.StandardMethod(scope.Service, scope.Resource, aipreflect.MethodTypeUpdate)
-		util.RangeRequiredFields(scope.Message.Desc, func(p protopath.Path, field protoreflect.FieldDescriptor) {
+		util.RangeRequiredFields(scope.Message.Desc, true, func(p protopath.Path, field protoreflect.FieldDescriptor) {
```
Comment (ericwenn) [inline internal/aiptest/update/required_fields.go:31]: Lets use `xrange.RangeFields` directly (`util.RangeRequiredFields` is just a wrapper around it), and then filter do the filtering in this method instead. "Naked" boolean arguments are hard to reason about in review..

Comment (radhus) [reply internal/aiptest/update/required_fields.go]: I opted for just checking the annotation here and ignoring it if present. WDYT? Not sure if we'd like to limit annotation checking to certain packages or similar, so feedback welcome again 🙏

Comment (ericwenn) [reply internal/aiptest/update/required_fields.go]: I like it!

**Review by ericwenn — APPROVED** [review]
Nice!

**Pull request merged**